internal-packages/next-dev/src/shared.ts

```diff
@@ -21,56 +35,64 @@ export function monkeyPatchVmModule({ env, cf, ctx, caches }: PlatformProxy) {
 
 	const originalRunInContext = vmModule.runInContext.bind(vmModule);
 
-	vmModule.runInContext = (
-		...args: [
-			string,
-			Record<string, unknown> & {
-				process?: { env?: Record<string, unknown> };
-				[cloudflareRequestContextSymbol]?: {
-					env: unknown;
-					ctx: unknown;
-					cf: unknown;
-				};
-			},
-			...[unknown],
-		]
-	) => {
+	vmModule.runInContext = (...args: [string, RuntimeContext, ...[unknown]]) => {
 		const runtimeContext = args[1];
 
-		if (!runtimeContext[cloudflareRequestContextSymbol]) {
-			runtimeContext[cloudflareRequestContextSymbol] = {
-				env,
-				ctx,
-				cf,
-			};
-			if (runtimeContext.process?.env) {
-				for (const [name, binding] of Object.entries(env)) {
-					runtimeContext.process.env[name] = binding;
-				}
-			}
+		runtimeContext[cloudflareRequestContextSymbol] ??= {
+			env,
+			ctx,
+			cf,
+		};
 
-			runtimeContext['caches'] = caches;
-
-			runtimeContext['Request'] = new Proxy(Request, {
-				construct(target, args, newTarget) {
-					if (
-						args.length >= 2 &&
-						typeof args[1] === 'object' &&
-						args[1].duplex === undefined
-					) {
-						args[1].duplex = 'half';
-					}
-					return Reflect.construct(target, args, newTarget);
-				},
-			});
-			runtimeContext['Response'] = Response;
-			runtimeContext['Headers'] = Headers;
-		}
+		monkeyPatchProcessEnv(runtimeContext, env);
+		monkeyPatchAuxiliaryGlobals(runtimeContext, caches);
 
 		return originalRunInContext(...args);
 	};
 }
 
+function monkeyPatchProcessEnv(
+	runtimeContext: RuntimeContext,
+	env: Record<string, unknown>,
+) {
+	if (
+		runtimeContext.process?.env &&
+		!runtimeContext.process.env[processEnvIsPatched]
+	) {
+		if (runtimeContext.process?.env) {
```

Comment:
This must always be true in here.

Reply:
yeah you're totally right, thanks for noticing it 😄 🙏